nixos/modules/hardware/video/nvidia.nix

```
@@ -275,7 +275,7 @@ in
               softdep nvidia post: nvidia-uvm
             '';
           };
-          systemd.tmpfiles.rules = lib.optional config.virtualisation.docker.enableNvidia "L+ /run/nvidia-docker/bin - - - - ${nvidia_x11.bin}/origBin";
+          systemd.tmpfiles.rules = lib.mkIf config.virtualisation.docker.enableNvidia "L+ /run/nvidia-docker/bin - - - - ${nvidia_x11.bin}/origBin";
```

Correct me if I'm wrong, but I think `lib.optional` wraps its result in a list (or returns an empty list) whereas `lib.mkIf` _assigns_ its result as-is (or assigns a lazy attr that doesn't exist(?)). `systemd.tmpfiles.rules` requires a list, so this causes an error when `virtualisation.docker.enableNvidia = true`.

```suggestion
          systemd.tmpfiles.rules = lib.mkIf config.virtualisation.docker.enableNvidia [ "L+ /run/nvidia-docker/bin - - - - ${nvidia_x11.bin}/origBin" ];
```

---

You're right 🤦🏼

---

#318924